### DISCUSSION TITLE
How to use FlxSave to save int?

### DISCUSSION THREAD

**Discussion by NikRetaNCAM** [HaxeFlixel/flixel#2474] [Q&A] [answered]
Let me explain
I have a variable named 'level'
![image](https://user-images.githubusercontent.com/60806448/147816408-a28c7c6e-bb16-46c0-9940-32e0b823cbfa.png)
That variable by default equals to 1
For each level you pass, the variable gets 1 more point (so like, you pass the first level and the variable turns 2 instead of 1)
![image](https://user-images.githubusercontent.com/60806448/147816449-7df70d23-20ce-428e-b9e6-a8121c0938fa.png)
![image](https://user-images.githubusercontent.com/60806448/147816486-ab66c1f6-8f12-4961-8c5d-1b7b46b69a4f.png)
When you open the game next time, i want to check for that save, and change the value of "level" to whatever the save has
Because in create() it is already written
![image](https://user-images.githubusercontent.com/60806448/147816353-b42f66ef-b7c3-45c0-9bb0-58b6434e58a8.png)
so if "level" is 2 for example, it loads different map

So the point of the question - how do i use FlxSave to actually save the int and load it then?

**Comment by Geokureli** [comment]
to access a `level` save file:
```hx
function getSavedLevel():Int
{
    // check null in case no level was saved yet
    if (FlxG.save.data.level != null)
        return FlxG.save.data.level;
    
    return 1;
}
```

and to save the level:
```hx
function saveLevel(level:Int)
{
    FlxG.save.data.level = level;
    // flush() actually saves whatever data you've set to local storage
    FlxG.save.flush();
}
```

Note that `FlxG.save.data` is dynamic, it can have any property you want and it'll work with just about any value, but it has no properties until you first save, so just be sure to null check when accessing primitive types or sub properties

**Reply by NikRetaNCAM** [reply]
> to access a `level` save file:
> 
> ```hx
> function getSavedLevel():Int
> {
>     // check null in case no level was saved yet
>     if (FlxG.save.data.level != null)
>         return FlxG.save.data.level;
>     
>     return 1;
> }
> ```
> 
> and to save the level:
> 
> ```hx
> function saveLevel(level:Int)
> {
>     FlxG.save.data.level = level;
>     // flush() actually saves whatever data you've set to local storage
>     FlxG.save.flush();
> }
> ```
> 
> Note that `FlxG.save.data` is dynamic, it can have any property you want and it'll work with just about any value, but it has no properties until you first save, so just be sure to null check when accessing primitive types or sub properties

may i ask: why in getSavedLevel() it returns 1?

edit: what's not right here?
![image](https://user-images.githubusercontent.com/60806448/147855455-80f8a4b0-600e-4465-b87f-d6e99089f4ff.png)
the game crashed ;-;

**Reply by Geokureli** [reply]
> may i ask: why in getSavedLevel() it returns 1?

one is the default value for when a player hasn't played and has no save file. I assume you want to player to start at level 1

> the game crashed ;-;

this is incredibly vague, where did you put the code, when is it executing, what crash info are you getting?

**Reply by NikRetaNCAM** [reply]
> > may i ask: why in getSavedLevel() it returns 1?
> 
> one is the default value for when a player hasn't played and has no save file. I assume you want to player to start at level 1
> 
> > the game crashed ;-;
> 
> this is incredibly vague, where did you put the code, when is it executing, what crash info are you getting?

all the code i shown is in PlayState
no crash info. i first enter MenuState, and then when it should transition to PlayState - the game just closes

**Reply by Geokureli** [reply]
there's nothing wrong with the parts I see, what happens if you comment it out calls to these functions, does it still crash? If not, try posting the whole class in https://gist.github.com/

**Reply by NikRetaNCAM** [reply]
> there's nothing wrong with the parts I see, what happens if you comment it out calls to these functions, does it still crash? If not, try posting the whole class in https://gist.github.com/

in a minute, hol' up

**Reply by NikRetaNCAM** [reply]
The game still crashes

what the-

**Reply by NikRetaNCAM** [reply]
Check this please
https://gist.github.com/NicklGames/86c38e97374073de5b22199cfbbe7a8b
what am i doing wrong?

**Comment by NikRetaNCAM** [comment, marked as answer]
ok so TURNS OUT
the game crashed bc i havent identified 'finish' as FlxSprite:
finish = new FlxSprite(0,0)

now i'll try to add saving/loading things
edit: it launches ok
